File: Service/GroupMembershipManagement/Models/Page.cs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

using System.Collections.Generic;

namespace Models
{
    public class Page<T>
    {
        public string Query { get; set; }
        public string ContinuationToken { get; set; }
        public IReadOnlyList<T> Values { get; set; } = new List<T>();
    }
}

File: UI/web-app/src/components/Page/Page.ts
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

import { styled } from '@fluentui/react'
import type * as React from 'react'

import { PageBase } from './Page.base'
import { getStyles } from './Page.styles'
import { type IPageProps, type IPageStyleProps, type IPageStyles } from './Page.types'

export const Page: React.FunctionComponent<IPageProps> = styled<IPageProps, IPageStyleProps, IPageStyles>(
  PageBase,
  getStyles,
  undefined,
  {
    scope: 'Page'
  }
)
